### PR TITLE
chain: Downgrade to Cosmos-SDK 0.38.4

### PR DESCRIPTION
**Pull request by sorawit**
In order to make sure we can control the release timeline, we decide to rollback BandChain to based on Cosmos-SDK 0.38.4. We will migrate to IBC version once a stable release has come.

### CODE REVIEW

File: chain/app/db_app.go [hunk 1]
```diff
@@ -38,7 +37,7 @@ func NewDBBandApp(
 	dbBand.DistrKeeper = app.DistrKeeper
 	dbBand.StakingKeeper = app.StakingKeeper
 	dbBand.OracleKeeper = app.OracleKeeper
-	dbBand.IBCKeeper = app.IBCKeeper
+	// dbBand.IBCKeeper = app.IBCKeeper
```
Comment (taobun) [inline chain/app/db_app.go]: Delete this

File: chain/app/db_app.go [hunk 2]
```diff
@@ -57,11 +56,13 @@ func (app *dbBandApp) InitChain(req abci.RequestInitChain) abci.ResponseInitChai
 	app.cdc.MustUnmarshalJSON(req.AppStateBytes, &genesisState)
 
 	// Bank balance genesis
-	var bankState bank.GenesisState
-	app.cdc.MustUnmarshalJSON(genesisState[bank.ModuleName], &bankState)
-
-	for _, account := range bankState.Balances {
-		err := app.dbBand.SetAccountBalance(account.Address, account.Coins, 0)
+	// var bankState bank.GenesisState
```
Comment (taobun) [inline chain/app/db_app.go]: Here too

**Review by Benzbeeb — APPROVED** [review]

**Review by perimeko — APPROVED** [review]

File: chain/cmd/bandoracled2/handler.go [hunk 3]
```diff
@@ -5,7 +5,7 @@ import (
 	"time"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
-	ibc "github.com/cosmos/cosmos-sdk/x/ibc/04-channel/types"
+	// ibc "github.com/cosmos/cosmos-sdk/x/ibc/04-channel/types"
```
Comment (taobun) [inline chain/cmd/bandoracled2/handler.go]: Delete!

File: chain/x/oracle/keeper/result.go [hunk 4]
```diff
@@ -50,7 +52,7 @@ func (k Keeper) GetAllResults(ctx sdk.Context) (results [][]byte) {
 	var previousReqID types.RequestID
 	defer iterator.Close()
 	for ; iterator.Valid(); iterator.Next() {
-		currentReqID := types.RequestID(sdk.BigEndianToUint64(iterator.Key()[1:]))
+		currentReqID := types.RequestID(binary.BigEndian.Uint64(iterator.Key()[1:]))
```
Comment (taobun) [inline chain/x/oracle/keeper/result.go:55]: Back to the basic lul

File: chain/x/supply/keeper.go [hunk 5]
```diff
@@ -1,67 +1,64 @@
...
-	ak          auth.AccountKeeper
 	distrKeeper *distr.Keeper
 }
 
-// WrapBankKeeperBurnToCommunityPool creates a new instance of WrappedBankKeeper
+// WrapSupplyKeeperBurnToCommunityPool creates a new instance of WrappedSupplyKeeper
 // with its distrKeeper member set to nil.
-func WrapBankKeeperBurnToCommunityPool(bk bank.Keeper, ak auth.AccountKeeper) WrappedBankKeeper {
-	return WrappedBankKeeper{bk, ak, nil}
+func WrapSupplyKeeperBurnToCommunityPool(bk supply.Keeper) WrappedSupplyKeeper {
```
Comment (taobun) [inline chain/x/supply/keeper.go]: sk

**Review by taobun — CHANGES_REQUESTED** [review]

**Review by taobun — APPROVED** [review]

**Pull request merged**